graphReconstruct/graphReconstruct.go

```diff
@@ -2,230 +2,89 @@ package graphReconstruct
-		SetState(node.Right, position)
-		if allZero(node.Left.Stored) && allZero(node.Right.Stored) {
-			log.Fatal("no Stored values passed to internal node")
-		}
-
-		for i := 0; i < 4; i++ {
-			sum := 0.0
-			for j := 0; j < 4; j++ {
-				for k := 0; k < 4; k++ {
-					sum = sum + Prob(i, j, node.Left.BranchLength)*node.Left.Stored[j]*Prob(i, k, node.Right.BranchLength)*node.Right.Stored[k]
-				}
-			} //branch length (transition probability) times the probability of a base appearing given the context of the lower tree nodes
-			node.Stored[i] = sum
-		}
-	} else if node.Left != nil {
-		SetState(node.Left, position)
-		if node.Left.Stored == nil {
-			log.Fatal("no Stored values passed to internal node, left branch")
-		}
-		for i := 0; i < 4; i++ {
-			sum := 0.0
-			for j := 0; j < 4; j++ {
-				sum = sum + Prob(i, j, node.Left.BranchLength)*node.Left.Stored[j]
-			} //branch length (transition probability) times the probability of a base appearing given the context of the lower tree nodes
-			node.Stored[i] = sum
-		}
-	} else if node.Right != nil {
-		SetState(node.Right, position)
-		if node.Right.Stored == nil {
-			log.Fatal("no Stored values passed to internal node, right branch")
-		}
-		for i := 0; i < 4; i++ {
-			sum := 0.0
-			for k := 0; k < 4; k++ {
-				sum = sum + Prob(i, k, node.Right.BranchLength)*node.Right.Stored[k]
-			}
-			node.Stored[i] = sum
-		}
-	} else if node.Right == nil && node.Left == nil {
-		node.State = 4 // starts as N
-		node.Stored = []float64{0, 0, 0, 0}
-
-		if len(node.Fasta.Seq) <= position {
-			log.Fatal("position specified is out of range of sequence \n")
-		} else if len(node.Fasta.Seq) > position {
-			node.State = int(node.Fasta.Seq[position])
-			for i := 0; i < 4; i++ {
-				if i == node.State {
-					node.Stored[i] = 1
-				} else {
-					node.Stored[i] = 0
-				}
+//BuildEdges connects the nodes of a species' graph that are stored in GraphColumns
+//func BuildEdges
+//start without prob
+//loop through species in column, go through all nodes
+//check if that node's seq matches the seq of the ancestor node without an edge
+//make the next of that node the same as this node's next
+//FindAncSeq creates a graph from the node records stored in GraphColumns and then calls PathFinder and seqOfPath to determine the most likley seq of the ancestor before assigning that
+//seq to the Fasta field of the ancestors tree node
+//func FindAncSeq will loop through aligncolumns and build a single graph of all of the nodes that belong to the ancestor species after edges are created
+//run PathFinder on the graph for the anc, run seqOfPath, then turn that to a fasta for that node of the tree
+
+//seqOfPath takes in a graph and a path specified by the Node IDs and returns the seq of the path through the graph
+func seqOfPath(g *simpleGraph.SimpleGraph, path []uint32) []dna.Base {
+	var seq []dna.Base
+	for n := 0; n < len(g.Nodes); n++ {
+		for p := 0; p < len(path); p++ {
+			if g.Nodes[n].Id == path[p] {
```

Comment:
I think we need a check to catch if this 'if' never evaluates to true for a given 'path[p]'.  This would then be an invalid path and should give an error.